src/hooks/useNodeViewPortals.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactPortal, useCallback, useState } from "react";
+import { createPortal } from "react-dom";
+
+import {
+  PORTAL_REGISTRY_ROOT_KEY,
+  PortalRegistryContext,
+  PortalRegistryKey,
+} from "../contexts/PortalRegistryContext.js";
+import { RegisterElement } from "../nodeViews/createReactNodeViewConstructor.js";
+
+import { useEditorState } from "./useEditorState.js";
+
+type Props = {
+  portals: Record<PortalRegistryKey, ReactPortal[]>;
+};
+
+function PortalTree({ portals }: Props) {
+  const editorState = useEditorState();
+  const rootPortals = editorState && portals[PORTAL_REGISTRY_ROOT_KEY];
```

Comment:
Why do we care about the editor state here?

Reply:
Oh, we don't anymore! Haha good catch